sktime/classification/feature_based/_adi_cv_classifier.py

```diff
@@ -0,0 +1,198 @@
+        "scitype:instancewise": False,
+        "scitype:transform-labels": "None",
+        "X_inner_mtype": "pd.DataFrame",
+        "y_inner_mtype": "None",
+        "univariate-only": True,  # Demand being the only variable
+        "requires_y": False,
+        "fit_is_empty": True,
+        "capability:inverse_transform": False,
+        "capability:unequal_length": False,
+        "handles-missing-data": False,
+        "authors": ["shlok191"],
+        "maintainers": ["shlok191"],
+    }
+
+    def __init__(self, adi_threshold=1.32, cv_threshold=0.49, features=None):
+        """Initialize the transformer and processes any provided parameters.
+
+        Parameters
+        ----------
+        adi_threshold : float, optional
+            Threshold for Average Demand Interval. Defaults to 1.32.
+
+        cv_threshold : float, optional
+            Threshold for Variance. Defaults to 0.49.
+
+        features : List[str] | None, optional
+            List of features to compute. Defaults to None (all features)
+
+        Raises
+        ------
+            ValueError: If features is provided and does not
+            contain 'adi','cv2', or 'class'.
+        """
+        self.adi_threshold = adi_threshold
+        self.cv_threshold = cv_threshold
+        self.features = features
+
+        # Initialize the parent class
+        super().__init__()
+
+        # Checking if the features parameter is valid
+        if features is not None:
+            if (
+                "adi" not in features
+                or "cv2" not in features
+                or "class" not in features
+            ):
+                raise ValueError(
+                    "The features list must either be None or include adi "
+                    + "cv2, and class as elements."
+                )
+
+        else:
+            # Helpful to transform None to default list for transform function
+            self.features = ["adi", "cv2", "class"]
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : Series
+            Series of time series data this transform function aims to classify
+
+        y : Series | None, default=None
+            Not required for the classification of the time series
+
+        Returns
+        -------
+        X_transformed : pd.DataFrame
+            Contains columns listed in self.features and 1 row entry of values
+
+        """
+        condition = X != 0
+        X_non_zero = X[condition]
+
+        # Calculating ADI value based on formula from paper
+        adi_value = len(X) - 1 / (len(X_non_zero) - 1)
+
+        # Calculating variance for all non-zero values
+        variance = X_non_zero.var()
+        cv2_value = variance / len(X_non_zero)
+
+        # Calculating the class type
+        class_type = ""
+
+        if adi_value <= self.adi_threshold:
```

Comment:
not blocking, but to write cleaner code, I would do the checks only twice, define boolean variables `adi_low` and `cv2_low`. Then do `if` / `elif` with four conditions from the boolean combinations.

Reply:
Got it! I'll update the code!